### PR TITLE
docs: document helm fileParameters / helm-set-file

### PR DESCRIPTION
**Pull request by cardoe**
Added documentation for fileParameters in a helm source Application or the --helm-set-file cli option from #2751. Added a note about #13220 so users are aware. #17941 aims to remove this limitation so the warning will be adjusted to mention the version that lands that support.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [x] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by cardoe** [comment]
It seemed like the documentation got missed in the original PR so this adds it. Should be possible to back port this to stable releases.

**Comment by cardoe** [comment]
Rebased this PR and updated the message to reference #17941 as the PR that implements the feature.

File: docs/user-guide/helm.md [hunk 1]
```diff
@@ -201,6 +201,28 @@ the result will be param1=value5
...
+```yaml
+source:
+    helm:
+        fileParameters:
+            - name: some.key
+              value: path/to/file.ext
+```
+
+!!! warning "Multiple sources not supported"
+    Please note that using a multiple sources application will not let you load the file by reference. See [argoproj/argo-cd#13220](https://github.com/argoproj/argo-cd/issues/13220)
```
Comment (crenshaw-dev) [inline docs/user-guide/helm.md:224]:
```suggestion
!!! warning "Reference in multiple sources not supported"
    Please note that using a multiple sources application will not let you load the file by reference. See [argoproj/argo-cd#13220](https://github.com/argoproj/argo-cd/issues/13220)
```

File: docs/user-guide/helm.md [hunk 2]
```diff
@@ -201,6 +201,28 @@ the result will be param1=value5
...
+```
+
+or using the fileParameters for yaml:
+
+```yaml
+source:
+    helm:
+        fileParameters:
+            - name: some.key
+              value: path/to/file.ext
```
Comment (crenshaw-dev) [inline docs/user-guide/helm.md:220]:
```suggestion
source:
  helm:
    fileParameters:
      - name: some.key
        value: path/to/file.ext
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by cardoe** [comment]
Thank you for the feedback. I've applied your changes.

**Review by crenshaw-dev — APPROVED** [review]
Thank you!

**Comment by laiminhtrung1997** [comment]
@cardoe @crenshaw-dev

**Pull request merged**